doc/source/admin/framework_dependencies.rst

```diff
@@ -0,0 +1,573 @@
+Galaxy now distributes dependencies in Python's new standard packaging format
+(Wheel), albeit still in a non-standard way. Thankfully, the new distribution
+method is far more compatible with the standard package distribution tooling
+(i.e. `pip`_) than the old method.
+
+.. _Python module dependencies: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/dependencies/requirements.txt
+.. _pip: https://pip.pypa.io/
+.. _wheel: https://wheel.readthedocs.org/
+
+How it works
+------------
+
+Upon startup (with ``run.sh``), the startup scripts will:
+
+1. Create a Python `virtualenv`_ in the directory ``.venv``.
+
+2. Unset the ``$PYTHONPATH`` environment variable (if set) as this can
+   interfere with installing `Galaxy pip`_ and dependencies.
+
+3. Replace that virtualenv's pip with `Galaxy pip`_.
+
+4. Download and install wheels from the Galaxy project wheel server,
+   `wheels.galaxyproject.org`_, using pip.
+
+5. Start Galaxy using ``.venv/bin/python``.
+
+.. _virtualenv: https://virtualenv.readthedocs.org/
+.. _wheels.galaxyproject.org: https://wheels.galaxyproject.org/
+
+Options
+-------
+
+A variety of options to ``run.sh`` are available to control the above behavior:
+
+- ``--skip-venv``: Do not create or use a virtualenv, and do not replace pip
+  with Galaxy pip.
+- ``--skip-wheels``: Do not install wheels.
+- ``--no-create-venv``: Do not create a virtualenv, but use one if it exists at
+  ``.venv`` or if ``$VIRTUAL_ENV`` is set (this variable is set by virtualenv's
+  ``activate``)
+- ``--replace-pip/--no-replace-pip``: Do/do not replace pip with Galaxy pip.
+
+Managing dependencies manually
+------------------------------
+
+Create a virtualenv
+^^^^^^^^^^^^^^^^^^^
+
+Using a `virtualenv`_ in ``.venv`` under the Galaxy source tree is not
+required. More complicated Galaxy setups may choose to use a virtualenv
+external to the Galaxy source tree, which can be done either by not using
+``run.sh`` directly (an example of this can be found under the `Supervisor`_
+section) or using the ``--no-create-venv`` option, explained in the `Options`_
+section. It is also possible to force Galaxy to start without a virtualenv at
+all, but you should not do this unless you know what you're doing.
+
+To manually create a virtualenv, you will first need to obtain virtualenv.
+There are a variety of ways to do this:
+
+- ``pip install virtualenv``
+- Install your Linux distribution's virtualenv package from the system package
+  manager (e.g. ``apt-get install python-virtualenv``).
+- Download the `virtualenv source from PyPI
+  <https://pypi.python.org/pypi/virtualenv>`_, untar, and run the
+  ``virtualenv.py`` script contained within as ``python virtualenv.py
+  /path/to/galaxy/virtualenv``
+
+Once this is done, create a virtualenv. In our example, the virtualenv will
+live in ``/srv/galaxy/venv`` and the Galaxy source code has been cloned to
+``/srv/galaxy/server``.
+
+.. code-block:: console
+    
+    $ virtualenv /srv/galaxy/venv
```

Comment:
The point of this section is to show creating a virtualenv in an alternate, admin-controlled location.